core/src/main/java/io/parsingdata/metal/Util.java

```diff
@@ -59,10 +61,21 @@ public static boolean notNullAndSameClass(final Object object, final Object othe
             && object.getClass() == other.getClass();
     }
 
-    public static String bytesToHexString(final byte[] bytes) {
+    public static String bytesToShortHexString(final byte[] bytes) {
         checkNotNull(bytes, "bytes");
-        final char[] hexChars = new char[bytes.length * 2];
-        for (int j = 0; j < bytes.length; j++) {
+        if (bytes.length > (MAX_VALUES_IN_HEX_STRING + 1)) {
+            final char[] hexChars = new char[(MAX_VALUES_IN_HEX_STRING * 2) + 3];
+            hexChars[hexChars.length-1] = '.';
```

Comment:
I would include the additional dots with `+ "..."` after the return, instead of these single sets in the array. This method can be simplified further, since you would not need a hexChar array for the fillHexDigits, only the count:

````
if (bytes.length > (MAX_VALUES_IN_HEX_STRING + 1)) {
    return fillHexDigits(bytes, MAX_VALUES_IN_HEX_STRING) + "...";
}
return fillHexDigits(bytes, bytes.length);
````